### PR TITLE
Update main configuration interface

### PR DESCRIPTION
**Pull request by patrickelectric**
Follow this changes: https://github.com/bluerobotics/ping-viewer/pull/359#issuecomment-447079705

### CODE REVIEW

File: src/settings/settingsmanager.h [hunk 1]
```diff
@@ -109,6 +109,8 @@ class SettingsManager : public QObject
     AUTO_PROPERTY(bool, replayMenu, false)
     AUTO_PROPERTY(bool, reset, false)
     AUTO_PROPERTY(bool, darkTheme, false)
+    AUTO_PROPERTY(bool, connectionGroup, false)
```
Comment (jaxxzer) [inline src/settings/settingsmanager.h]: showConnectionOptions?

File: src/settings/settingsmanager.h [hunk 2]
```diff
@@ -109,6 +109,8 @@ class SettingsManager : public QObject
...
+    AUTO_PROPERTY(bool, sensorAdvancedConfiguration, false)
```
Comment (jaxxzer) [inline src/settings/settingsmanager.h]: showAdvancedSensorOptions?

File: src/sensor/ping.h [hunk 3]
```diff
@@ -59,7 +59,46 @@ class Ping : public Sensor
...
      */
     uint8_t srcId() { return _srcId; }
-    Q_PROPERTY(int srcId READ srcId NOTIFY srcIdUpdate)
+
+    /** TODO: This is not working
+     * @brief Set device id
+     *
+     * @param id
+     */
+    void srcId(int id)
```
Comment (jaxxzer) [inline src/sensor/ping.h]: I think maybe we should refactor this to device_id

srcID is in context of a message. Here we are representing state of a device.

**Comment by jaxxzer** [comment]
I think it will be better if the effect of the enable/disable click can be immediately realized. Right now, the user enables it, then has to go find it. If the option and the effect are in the same visual context, it will be more clear and easy to use.

Perhaps a dropdown/expand menu sort of arrow with heading:

Ping main settings
[ always show]
Ping advanced settings >
Connection settings > <--- click row/arrow to expand menu immediately below.

File: src/sensor/ping.h (hunk 3 above)
Comment (patrickelectric) [reply src/sensor/ping.h]: oooops, this should not be here.

**Comment by patrickelectric** [comment]
@jaxxzer 	new layout.
deepin-screen-recorder_select area_20181219092659

**Pull request merged**